### DISCUSSION TITLE
q-input textarea

### DISCUSSION THREAD

**Discussion by thlinde** [quasarframework/quasar#8778] [General - Components / Directives / etc] [answered]
Hi all, i want to progammatically insert text in a textarea at the last cursor position. With Quasar 1 / Vue 2 i can get it with `this.$refs.textAreaText.$refs.input`, where textAreaText is a ref in template.
In Quasar 2 with composition api i have in the setup function `const textAreaText = ref('')`, return it for the ref in the template, but get with `textAreaText.value` no possibility to access cursor position.
What I am doing wrong. Can someone hplease help? Thank you all, thlinde.

**Comment by hawkeye64** [comment, marked as answer]
@thlinde Maybe this link will help: https://stackoverflow.com/a/7745998/5359558

**Reply by thlinde** [reply]
Thank you hawkeye64 for the prompt reply. The problem is not to find the cursor position in the textarea, but i get no reference to the q-input element with Quasar 2 (or maybe Vue 3), that i have in Quasar 1.
With `textAreaText.value` i get no object to use element.selectionStart (undefined). My question is how to get this element with `<q-input ref="textAreaText" />` in the script part, so that i can use `textAreaText.value.selectionStart`. Sorry for the complicated explanation of the question. So thank you again.

**Reply by thlinde** [reply]
Ok, when i use the pure html component textarea with some styling all works as expected. Maybe there is a problem of the quasar component when using it with template ref!?

**Reply by hawkeye64** [reply]
Not sure about this, but if using it as a `ref=...`, then set the initial state of the ref to null `ref(null)` and not `ref('')`

**Reply by thlinde** [reply]
Hi Jeff. I've done a little demo in jsfiddle that shows the problem. It would be nice if you have time to look at it:
http://jsfiddle.net/whsdazr0/14/

**Reply by metalsadman** [reply]
> Hi Jeff. I've done a little demo in jsfiddle that shows the problem. It would be nice if you have time to look at it:
> http://jsfiddle.net/whsdazr0/14/

you can use `querySelector` for now, `qinputRef.value.$el.querySelector('textarea').selectionStart`.
http://jsfiddle.net/vh0a978j/

**Reply by thlinde** [reply]
Thank you all. That makes it clear now. You make a great work with the Quasar framework. Best regards.

**Comment by labs20** [comment]
I found this function `getNativeElement()` avaiable on the q-input and it returns the native html input inside of it. You need this if you want to manipulate selectionStart.
